src/NServiceBus.Core/Persistence/Development/SagaPersister/DevelopmentSagaIdGenerator.cs

```diff
@@ -0,0 +1,16 @@
+namespace NServiceBus.Features
+{
+    using System;
+    using Extensibility;
+    using NServiceBus.Sagas;
+
+    class DevelopmentSagaIdGenerator : ISagaIdGenerator
+    {
+        public Guid Generate(string propertyName, object propertyValue, SagaMetadata metadata, ContextBag context)
+        {
+            //here we assume single sagas since v6 doesn't allow more than one corr prop
+            // note that we still have to use a GUID since moving to a string id will have to wait since its a breaking change
+            return DeterministicGuid.Create(propertyValue);
```

Comment:
just thinking through some edge cases:
Two Saga types `Saga1` and `Saga2` both handling a `TimeoutMessage` via `RequestTimeout`. In case `Saga1` requests a timeout, how is ensured that only saga1 actually handles the timeout?
(tl;dr I think the ID should be unique across saga types)